### DISCUSSION TITLE
Unable to convert DMRed PDF from ADE [SOLVED]

### DISCUSSION THREAD

**Discussion by trafra99** [apprenticeharper/DeDRM_tools#1627] [Q&A]
Hi guys! I haven't managed to convert a PDF downloaded with ADE, I have attached the debug log so that hopefully someone can help me 🙏  
Edit: problem solved thanks to @ElleKayEm 🙏🏻

[tmpez77fl3x.txt]
(https://github.com/apprenticeharper/DeDRM_tools/files/6216153/tmpez77fl3x.txt)

**Comment by ElleKayEm** [comment]
Have you tried calibre 4.23 with DeDRM 6.8.1?

**Reply by trafra99** [reply]
Thanks for the feedback, I'm gonna try right now

**Reply by trafra99** [reply]
> Have you tried calibre 4.23 with DeDRM 6.8.1?
@ElleKayEm 

I reloaded the file into Calibre an now it's working, but the outputs are all distorted, i tried with epub and pdf but it looks unreadable

**Reply by ElleKayEm** [reply]
If you're using the calibre ebook viewer, I'd try something else.

**Reply by trafra99** [reply]
> If you're using the calibre ebook viewer, I'd try something else.

I am using mac native pdf viewer and I've also tried other tools but the result is always the same. Images are spread without any criteria and so are the texts. I'll attach an example of how it is vs how it should be.
![Schermata 2021-03-27 alle 23 56 21](https://user-images.githubusercontent.com/81485665/112737088-1f4d8480-8f58-11eb-95d7-9b83af9aa6de.png)
![Schermata 2021-03-27 alle 23 56 41](https://user-images.githubusercontent.com/81485665/112737089-207eb180-8f58-11eb-84a5-978caf62d150.png)

**Reply by ElleKayEm** [reply]
This is DRM removal only?  Before any conversion attempt?

**Reply by trafra99** [reply]
> This is DRM removal only? Before any conversion attempt?

I put the DRMed original PDF into the console, then I converted it in different formats but I’m the same problem.

**Reply by ElleKayEm** [reply]
What about the DRM-free PDF?  Just after adding to calibre and without trying to convert?  PDFs are one of the worst formats to convert from or to.

**Reply by trafra99** [reply]
> What about the DRM-free PDF? Just after adding to calibre and without trying to convert? PDFs are one of the worst formats to convert from or to.

If I got it right you mean opening the file from the book list without doing anything, just after copying the pdf from the ade folder to calibre. 
If I try to do so calibre won’t let me open the file

**Reply by ElleKayEm** [reply]
Should have DRM removed as soon as you add it.  Try removing and re-adding the book to calibre.

**Reply by trafra99** [reply]
> Should have DRM removed as soon as you add it. Try removing and re-adding the book to calibre.


I did it! I managed to open the file and save a copy on the desktop, and it was drm free 😍 no conversion was needed as you said.
You’re a hero! I mean it. Thank you so so so much. 🙏🏻🙏🏻🙏🏻🙏🏻

**Reply by ElleKayEm** [reply]
Glad to hear it!
